### ISSUE TITLE
Why does the learning rate suddenly increase

### ISSUE THREAD

**Issue by Gary55555** [ultralytics/yolov5#12569]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello,
    Why does my learning rate suddenly increase when I reach approximately 1739 epochs during training?As a result, my model loss has also increased.
    My hyperparameters:
hyp:
  lr0: 0.01
  lrf: 0.01
  momentum: 0.937
  weight_decay: 0.0005
  warmup_epochs: 3.0
  warmup_momentum: 0.8
  warmup_bias_lr: 0.1
  box: 0.05
  cls: 0.5
  cls_pw: 1.0
  obj: 1.0
  obj_pw: 1.0
  iou_t: 0.2
  anchor_t: 4.0
  fl_gamma: 0.0
  hsv_h: 0.015
  hsv_s: 0.7
  hsv_v: 0.4
  degrees: 0.0
  translate: 0.1
  scale: 0.5
  shear: 0.0
  perspective: 0.0
  flipud: 0.0
  fliplr: 0.5
  mosaic: 1.0
  mixup: 0.0
  copy_paste: 0.0
epochs: 5000
batch_size: 1024
imgsz: 640
rect: false
resume: true
nosave: false
noval: false
noautoanchor: false
noplots: false
evolve: null
bucket: ''
cache: null
image_weights: false
device: 0,1,2,3,4,5,6,7
multi_scale: false
single_cls: false
optimizer: SGD
sync_bn: false
workers: 64
project: runs/train
name: exp
exist_ok: false
quad: false
cos_lr: false
label_smoothing: 0.0
patience: 100
freeze:
- 0
save_period: -1
seed: 0
local_rank: -1
entity: null
upload_dataset: false
bbox_interval: -1
artifact_alias: latest
save_dir: runs/train/exp9

**Comment by Gary55555** [comment]
![微信图片_20240102205855](https://github.com/ultralytics/yolov5/assets/35791309/77fc8c99-0932-45a1-b59a-648113859827)

**Comment by glenn-jocher** [comment]
@Gary55555 hello! It looks like you're experiencing an unexpected increase in learning rate during training. This could be due to a few reasons, such as a misconfiguration of the learning rate scheduler or an issue with the state of the optimizer if you're resuming training from a checkpoint.

Given that you're resuming training (`resume: true`), please ensure that the optimizer state is being correctly loaded along with the model weights. If the state is not loaded properly, the learning rate scheduler might reset or behave unexpectedly.

If everything seems to be configured correctly, consider whether any custom modifications to the training loop or learning rate scheduler could be causing this issue. If you're using a custom learning rate scheduler, ensure it's implemented correctly.

For further assistance, please provide more details on the learning rate scheduler you're using and any relevant logs that show the learning rate progression over epochs. You can also refer to the Ultralytics Docs for more information on configuring training hyperparameters and understanding the expected behavior of the learning rate scheduler.

Keep in mind that the learning rate is a crucial hyperparameter, and its schedule should be carefully set to ensure stable and effective training. If you continue to face issues, consider experimenting with different learning rate schedules or hyperparameters.

Thank you for reaching out, and good luck with your training! 🚀
